source/grid/grid_tools.cc

```diff
@@ -5962,6 +5962,39 @@ namespace GridTools
 
   namespace internal
   {
+    std::tuple<std::vector<unsigned int>,
+               std::vector<unsigned int>,
+               std::vector<unsigned int>>
+    convert_guessed_ranks_and_indices_to_crs(
+      std::vector<std::pair<unsigned int, unsigned int>> &ranks_and_indices)
+    {
+      std::sort(ranks_and_indices.begin(), ranks_and_indices.end());
+
+      std::vector<unsigned int> ranks;
+      std::vector<unsigned int> ptr;
+      std::vector<unsigned int> indices;
+
+      unsigned int dummy_rank = numbers::invalid_unsigned_int;
+
+      for (const auto &i : ranks_and_indices)
+        {
+          if (dummy_rank != i.first)
+            {
+              dummy_rank = i.first;
```

Comment:
This last assignment makes the `dummy_rank` no longer a "dummy" :-) I would call the variable `current_rank`.